test/test_consumer.py

```diff
@@ -15,11 +15,15 @@
 class TestKafkaConsumer:
     def test_session_timeout_larger_than_request_timeout_raises(self):
         with pytest.raises(KafkaConfigurationError):
-            KafkaConsumer(bootstrap_servers='localhost:9092', api_version=(0,9), group_id='foo', session_timeout_ms=60000, request_timeout_ms=40000)
+            KafkaConsumer(bootstrap_servers='localhost:9092', api_version=(0, 9), group_id='foo', session_timeout_ms=50000, request_timeout_ms=40000)
 
     def test_fetch_max_wait_larger_than_request_timeout_raises(self):
         with pytest.raises(KafkaConfigurationError):
-            KafkaConsumer(bootstrap_servers='localhost:9092', fetch_max_wait_ms=41000, request_timeout_ms=40000)
+            KafkaConsumer(bootstrap_servers='localhost:9092', fetch_max_wait_ms=50000, request_timeout_ms=40000)
+
+    def test_request_timeout_larger_than_connections_max_idle_ms_raises(self):
+        with pytest.raises(KafkaConfigurationError):
+            KafkaConsumer(bootstrap_servers='localhost:9092', api_version=(0, 9), request_timeout_ms=50000, connections_max_idle_ms=40000)
```

Comment:
I do not understand why this test fails with `NoBrokerFound` if `api_version=(0, 9)` is omitted... or rather, I realize it's a problem with the version probing timing out, but why does `test_fetch_max_wait_larger_than_request_timeout_raises()` right above succeed without needing to specify the version?

Reply:
I think that may have been a side effect of the old AND check. If you drop the api_version here I would expect the test to still pass

Reply:
IIRC, I actually did try that exact thing this morning, and it still failed with the `NoBrokerFound` error. That's what confuses me.